### PR TITLE
Add default azure credential sample for eventhubs

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
As subject says.
Forked from prior PR so that EH and Blob samples can go in distinctly.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhubs/examples/default_azure_credential_auth.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# For example user to be logged in can be specified by the environment variable AZURE_USERNAME, consumed via the ManagedIdentityCredential
+# Alternately, one can specify the AZURE_TENANT_ID, AZURE_CLIENT_ID, and AZURE_CLIENT_SECRET to use the EnvironmentCredentialClass.
+# The docs above specify all mechanisms which the defaultCredential internally support.
+
+credential = DefaultAzureCredential()
+client = EventHubClient(host=HOSTNAME,
+                        event_hub_path=EVENT_HUB,
+                        credential=credential)
+
+producer = client.create_producer(partition_id='0')
```
Comment (YijunXieMS) [inline sdk/eventhub/azure-eventhubs/examples/default_azure_credential_auth.py]: This looks like a track2 sample but not the latest version of track2 sdk.
Track 1 has client.add_producer, not create_

**Comment by yunhaoling** [comment]
Hi @KieranBrantnerMagee , two things worth to note.

1. we have renamed folder `examples` to `samples`.
2. we removed `EventHubClient`, for sending it's EventHubProducerClient.

I'm already put an identity sample [here](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventhub/azure-eventhubs/samples/sync_samples/client_secret_auth.py). 
Maybe you could update that sample to include the `DefaultAzureCredential`?

File: sdk/eventhub/azure-eventhubs/examples/default_azure_credential_auth.py (hunk 1 above)
Comment (KieranBrantnerMagee) [reply sdk/eventhub/azure-eventhubs/examples/default_azure_credential_auth.py]: Thanks for the callout.  (for you and Adam below).

I'll adjust the APIS, move to samples, and align it with the identity samples yall already wrote.

**Comment by KieranBrantnerMagee** [comment]
@yunhaoling Have adjust the PR as we discussed, largely integrating this sample into the client_identity sample yall wrote.  Let me know if you think this is forcing too much content into once place, I can always split it out into its own file/rewrite/whatever.

(To avoid dealing with reverts/etc for a change that's almost entirely pivoting, I've put this into a new PR, [here](https://github.com/Azure/azure-sdk-for-python/pull/9342) and am closing this)

**Pull request closed without merging**